src/peripherals.rs

```diff
@@ -10,6 +10,7 @@ use crate::mac;
 use crate::modem;
 #[cfg(any(esp32, esp32s2, esp32s3, esp32c6))]
 use crate::pcnt;
+#[cfg(any(feature = "rmt-legacy", esp_idf_version_major = "4"))]
```

Comment:
As per above, remove the `cfg` line. The code in the `esp_idf_hal::rmt` module should be changed so that the RMT _peripherals_ are always defined. Even if the legacy RMT driver is `#ifNdef`-ed.

Reply:
Done